popup.js

```diff
@@ -118,7 +157,24 @@ function handleClick(tab, url) {
     return getCsrfToken(url.origin).then(token => handleTokenResult(tab, url, token));
 }
 
+function fetchMessagesForLang(lang) {
+    return fetch(browser.extension.getURL(`i18n/${lang}.json`));
+}
+
+function geti18nMessages(preferredLang) {
+    let lang = preferredLang;
+    return fetchMessagesForLang(lang).then(res => res.json())
+    .catch(err => {
```

Comment:
Could get the fallback chain from siteinfo and use that instead.

Reply:
Since this extension is more a part of the browser chrome than part of a wiki site, it makes more sense to me to use the browser UI language when displaying messages, rather than the language of the wiki site content being viewed.  Any resemblance to wiki language fallback chains is coincidental!

Actually, this fallback chain was my first stab at attempting to translate between the [RFC 5646](https://tools.ietf.org/html/rfc5646) language tag provided by the browser via [browser.i18n.getUILanguage()](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/API/i18n/getUILanguage), and the ISO 639 language codes used in MediaWiki. 

Heuristic:
(1) Just try the value provided browser.i18n.getUILanguage() first; 
(2) If that's not found, search by just the first (primary) segment; 
(3) And then fall back to English (which also happens to be the language of the content provided by default for untranslated messages, at least on the wiki I'm requesting from).

Reply:
Huh, there's an entire RFC on just this subject. https://www.ietf.org/rfc/rfc4647.txt

Reply:
Hmm, OTOH, maybe using the wiki site language (with appropriate fallbacks, perhaps added later) is more sane overall even if less correct from a UI perspective, since I won't have to get into code format translation as outlined above.  That seems like a path to madness.

Reply:
OK, updated to use the wiki language for sanity's sake.

Reply:
> Alternatively, if you don't care too much about language support for anonymous users (reading lists can't be used anonymously, after all), you could just fetch the messages on the fly. That will use the interface language for logged-in users, which is arguably even better than the browser UI language (and definitely better than the content language).

Thanks for this suggestion.  Letting the API handle this seems like the best option of all, so I updated to do this (and just bundle English strings to fall back if this fails for whatever reason).